---
title: "mel_vcf_cleaning_haploidformat"
output: html_document
date: '2023-04-14'
---

```{r}
library(tidyverse)
```

# Quality control

```{r}
mel_raw0 <- read.csv("full_genome_filtered2.csv")
#removing low quality sites
mel_raw0$QUAL <- as.integer(mel_raw0$QUAL)
mel_raw00 <- mel_raw0 %>% 
  filter(QUAL > 1000)
#removing sites with alignment issues causing multiple alternates
mel_raw00 <- mel_raw00 %>% 
  filter(!grepl(",", ALT))

mel_raw00[] <- lapply(mel_raw00, function(x) gsub(".:0,0", "missing:missing:0", x))

mel_filtered0 <- mel_raw00 %>%
  mutate(across(c(CdBM23_S1:ZnBM47_S95), ~ map_chr(str_split(., ":"), 3), .names = "{col}_c")) %>% 
  mutate(across(ends_with("_c"), ~as.numeric(.))) %>% 
  mutate(across(c(CdBM23_S1:ZnBM47_S95), ~ map_chr(str_split(., ":"), 1), .names = "{col}_a")) %>% 
  select(-c(CdBM23_S1:ZnBM47_S95))

mel_filtered <- mel_filtered0 %>% 
  select(names(mel_filtered0)[1:9], sort(names(mel_filtered0)[-(1:9)])) %>% 
  mutate(across(ends_with("_a"), ~if_else(get(str_replace(cur_column(), "_a", "_c")) < 5, "low coverage", .))) %>% 
  select(-contains("_c"))

colnames(mel_filtered)<-gsub("_a","",colnames(mel_filtered))
```

## Cleaning up data

```{r}
#I need to change the chromosome name format to something easier to read
mel_chr <- mel_filtered %>% 
  mutate(chr=case_when(grepl("001133", CHROM) ~"chr1",
                       grepl("001134", CHROM) ~"chr2",
                       grepl("001135", CHROM) ~"chr3",
                       grepl("001136", CHROM) ~"chr4",
                       grepl("001137", CHROM) ~"chr5",
                       grepl("001138", CHROM) ~"chr6",
                       grepl("001139", CHROM) ~"chr7",
                       grepl("001140", CHROM) ~"chr8",
                       grepl("001141", CHROM) ~"chr9",
                       grepl("001142", CHROM) ~"chr10",
                       grepl("001143", CHROM) ~"chr11",
                       grepl("001144", CHROM) ~"chr12",
                       grepl("001145", CHROM) ~"chr13",
                       grepl("001146", CHROM) ~"chr14",
                       grepl("001147", CHROM) ~"chr15",
                       grepl("001148", CHROM) ~"chr16",
                       grepl("001224", CHROM) ~"chrmt"),
         .after = CHROM) %>% 
  select(-CHROM)

#Now I have to reorder the levels so they're not in the default alphabetical order
mel_chr$chr <- as.factor(mel_chr$chr)
levels(mel_chr$chr)
mel_chr$chr <- factor(mel_chr$chr, levels=c("chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11", "chr12", "chr13", "chr14", "chr15", "chr16", "chrmt"))

#Checking to see if I can delete some unecessary columns. They're unecessary because they have the same value in every row.
levels(mel_chr$ID)
levels(mel_chr$FILTER)

#removing columns where values in all rows are the same
mel_chr <- mel_chr %>% 
  select(-ID) %>% 
  select(-FILTER) %>% 
  select(-FORMAT)

cols_to_check0 <- colnames(mel_chr)
cols_to_check <- cols_to_check0[-(1:7)]
num_lines <- as.data.frame(cols_to_check) %>% 
  mutate(metal = case_when(grepl("Cd", cols_to_check) ~"Cd",
                           grepl("Co", cols_to_check) ~"Co",
                           grepl("Cu", cols_to_check) ~"Cu",
                           grepl("Mn", cols_to_check) ~"Mn",
                           grepl("Ni", cols_to_check) ~"Ni",
                           grepl("Zn", cols_to_check) ~"Zn",
                           grepl("OLY", cols_to_check) ~"ancestor",)) %>% 
  group_by(metal) %>% 
  count()
```

```{r}
#further filtering
#let's look at the quality 
hist(log10(mel_raw0$QUAL), breaks=100, main="", xlab="log(Genotype Quality)")
abline(v=log10(1000), col="red")
#I'm gonna filter out QUAL<1000 at the beginning
```

## Cleaning up data part 2

```{r}
#Now I will remove any sites where all lines have the same genotype
mel_RMancestral <- mel_chr %>%
  filter_at(vars(cols_to_check), any_vars((. != CdBM23_S1)))

#This is removing all sites where there are no instances of 0

#mel_subset0 <- mel_RMancestral[apply(mel_RMancestral[, 10:116], 1, function(x) any(grepl("0", x))), ]

mel_RMancestral <- mel_RMancestral %>%
  filter(rowSums(across(everything(), ~ str_detect(., "low coverage"))) <= 50)

#This is changing all data that matches the OLY077 column to "w303" so they can be marked as ancestral
match_col <- "OLY077_S101"
mel_RMancestral[] <- lapply(mel_RMancestral, function(x) ifelse(x == mel_RMancestral[[match_col]], "w303", x))

mel_RMancestral$chr <- sub("^", "chr", mel_RMancestral$chr)
mel_RMancestral[] <- lapply(mel_RMancestral, function(x) gsub("chr17", "chrmt", x))

#This is removing all sites where there are no instances of 1|1
mel_subset <- mel_RMancestral[apply(mel_RMancestral[, 10:116], 1, function(x) any(grepl("1", x))), ]
mel_subset_test <- mel_subset[apply(mel_subset[, 10:116], 1, function(x) any(grepl("0", x))), ]



#This is to get what sites to annotate
sites_to_check0 <- data.frame(matrix(nrow=268,ncol=1))
sites_to_check0$key <- paste(mel_subset$chr, mel_subset$POS, sep="_")
sites_to_check <- sites_to_check0 %>% 
  select(key) %>% 
  separate(key, into = c("chr", "POS"), sep="_") 
sites_to_check %>% 
  write.csv("sites_list_subset0.csv")
  
```

# Annotating genes
no00remove has 79 evo lines
no00keep has 107 evo lines

```{r}
annot <- read.csv("sites_list_subset_ANNOT.csv") %>% 
  rename("chr"="CHR") %>% 
  arrange(chr, POS)

mel_subset$POS <- as.integer(mel_subset$POS)

mel_mut <- mel_subset %>% 
  pivot_longer(cols=CdBM23_S1:ZnBM47_S95, names_to = "evo_line", values_to = "genotype") %>% 
  filter(genotype=="1") %>% 
  mutate(metal = case_when(grepl("Cd", evo_line) ~"Cd",
                           grepl("Co", evo_line) ~"Co",
                           grepl("Cu", evo_line) ~"Cu",
                           grepl("Mn", evo_line) ~"Mn",
                           grepl("Ni", evo_line) ~"Ni",
                           grepl("Zn", evo_line) ~"Zn")) %>% 
   arrange(chr,POS,metal)

mel_mut_annot <- left_join(mel_mut, annot, by = c('chr', 'POS')) %>% 
  #replace(is.na(.), 0) %>% 
  select(-c(FUNCTION:X.7)) %>% 
  filter(GENE!="NAG") %>% 
  relocate(GENE, .after=ALT) %>% 
  # These next two steps remove unknown or putative areas
  filter(!str_starts(GENE, "-")) %>% 
  filter(!str_starts(GENE, " -")) %>% 
  # These next few steps are removing genes from families that cause alignment issues
  filter(!grepl("FLO1", GENE)) %>% 
  filter(!grepl("FLO9", GENE)) %>% 
  #filter(!grepl("PIR3", GENE)) %>% 
  #filter(!grepl("ARS319", GENE)) %>% 
  #filter(!grepl("TIR1", GENE)) %>% 
  # reorder columns
  relocate(gene_code, .before = GENE) %>% 
  relocate(metal, .before = evo_line) %>% 
  select(-genotype) #%>% 
 # write.csv("mutations_haploid.csv")

mel_mut_annot$GENE_abbrev <- str_extract(mel_mut_annot$GENE, "^.*(?= - )")
mel_mut_annot <- mel_mut_annot %>% 
  relocate(GENE_abbrev, .before=GENE)

mel_mut_annot %>% 
  filter(evo_line!="MnBM42_S84") %>% 
  ggplot(aes(x=GENE_abbrev, fill=metal)) +
  geom_bar(stat="count") +
  theme(axis.text.x = element_text(angle=90))

mel_mut_annot$SNP <- paste(mel_mut_annot$GENE_abbrev, "_", mel_mut_annot$POS)

mel_mut_annot %>% 
  filter(evo_line!="MnBM42_S84") %>% 
  ggplot(aes(x=SNP, fill=metal)) +
  geom_bar(stat="count") +
  scale_fill_manual(values=c("#FF9A17", "#3B83FF", "#3EFF3B", "#FD73FF", "#FFF200", "#A880FF")) +
  theme(axis.text.x = element_text(angle=90))

distinct_genes <- mel_mut_no00remove_annot %>% group_by(GENE) %>% summarise(distinct_genes = n_distinct(GENE)) %>% 
  select(GENE) 
distinct_genes$GENE_abbrev <- str_extract(distinct_genes$GENE, "^.*(?= - )")
distinct_genes %>% write.csv("distinct_genes_5DP.csv")

```

####TEST SITE

```{r}
mel_mut_annot_wide <- mel_mut_annot %>% 
  filter(evo_line!="MnBM42_S84") %>% 
  pivot_wider(id_cols=c(chr:INFO), names_from = evo_line, values_from = metal)

mel_mut_no00remove_wide <- mel_mut_no00remove %>% pivot_wider(id_cols=c(chr:INFO), names_from = evo_line, values_from = metal)
```
